addons/edi_sale_notifier/tests/test_sale_notifier.py

```diff
@@ -0,0 +1,80 @@
+from odoo.addons.edi_sale.tests.common import EdiSaleCase
+from odoo import fields
+
+from unittest.mock import patch
+
+
+class TestSaleNotifier(EdiSaleCase):
+    """EDI sale order request tests"""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.sale_request_tutorial_record_type = cls.env.ref(
+            "edi_sale.sale_request_tutorial_record_type"
+        )
+        cls.sale_line_request_tutorial_record_type = cls.env.ref(
+            "edi_sale.sale_line_request_tutorial_record_type"
+        )
+        cls.doc_type_tutorial = cls.env.ref("edi_sale.sale_request_tutorial_document_type")
+        cls.doc_type_tutorial.active = True
+        cls.sale_request_tutorial_record_type.active = True
+        cls.sale_line_request_tutorial_record_type.active = True
+
+    @classmethod
+    def create_tutorial(cls, *filenames, fail_fast=True):
+        """Create sale order request tutorial document"""
+        cls.doc_type_tutorial.fail_fast = fail_fast
+        return cls.create_input_document(cls.doc_type_tutorial, *filenames)
+
+    def prepare_document(self, doc):
+        """Actions taken when preparing document"""
+        SaleRequestTutorialRecord = self.env["edi.sale.request.tutorial.record"]
+
+        res = doc.action_prepare()
+        # Find related records and fill in client_order_ref and requested_date
```

Comment:
Add a TODO here to resolve once we've reviewed the `udes_open` CI Dockerfile.

Reply:
Done